pkg/installation/install.go

```diff
@@ -76,7 +77,17 @@ func Install(p environment.Paths, plugin index.Plugin, forceDownloadFile string)
 	if err != nil {
 		return err
 	}
-	return install(plugin.Name, version, uri, bin, p, fos, forceDownloadFile)
+
+	glog.V(2).Infof("Storing install receipt for plugin %s", plugin.Name)
+	if err = receipt.Store(plugin, p.PluginReceiptPath(plugin.Name)); err != nil {
+		return errors.Wrap(err, "installation receipt could not be stored, uninstall may fail")
+	}
+
+	// The actual install should be the last action so that a failure during receipt
+	// saving does not result in an installed plugin without receipt.
+	glog.V(3).Infof("Install plugin %s", plugin.Name)
+	err = install(plugin.Name, version, uri, bin, p, fos, forceDownloadFile)
```

Comment:
[P1] looks like upgrading a plugin doesn't save an install receipt.
we should probably fix that too.

Reply:
That is definitely a :beetle: . I'll look into this.